### ISSUE TITLE
Gamepad Nav Scrolling outside of inner-rect bounds flickers and loses current navId value for a frame.

### ISSUE THREAD

**Issue by JohnnyonFlame** [ocornut/imgui#6171]
**Version/Branch of Dear ImGui:**
commit 752603bc70bc0fc24846f27b816f04c9fc356453 (HEAD -> master, origin/master, origin/HEAD)

**Back-end/Renderer/Compiler/OS**
SDL2, OpenGL ES2.x/GL 3.x

Back-ends: _imgui_impl_opengl3.cpp, imgui_impl_sdl2.cpp_
Compiler: GCC 11.2.0 (_arm-linux-gnueabihf_)
Operating System: Linux 4.4.x (_buildroot_)

**My Issue/Question:**

Whenever I hold the navigation button on a gamepad on a long list, and the nav goes outside of the inner rect, the current navId gets cleared by `ImGui::NavUpdateCreateMoveRequest()`. This causes certain menus I have to lose the nav cursor for a frame and fail to work as expected.

I have uncommented (as seen in the gist on the "verifiable example" list)  `IMGUI_DEBUG_LOG_NAV("[nav] NavMoveRequest: clamp NavRectRel for gamepad move\n");` and whenever the problem occurs, said snippet is triggered, showing the nav problem happening in real time, commenting `//g.NavId = 0;` under that causes the problem to be worked around.

**Standalone, minimal, complete and verifiable example:**

https://gist.github.com/JohnnyonFlame/7b5bb7de03ca8b92bcb6aba6df1d3854

**Comment by ocornut** [comment]
Hello Johnny,

I looked into this and tried to use your code and couldn't repro your issue.
Given the effort you put into formatting the gist (thank you) can you confirm that running this exact code you get the issue? (and not this code with some more code you haven't published). 

>  This causes certain menus I have to lose the nav cursor for a frame and fail to work as expected.

Could you take a video capture of it?

My guess was that you might have some other code interfering with NavId.

The log entry you uncommented triggers when pressing a nav direction on the gamepad while the previous entry is outside the view. It is desirable in this condition but shouldn't happen when scrolling a continuous list.

**Comment by JohnnyonFlame** [comment]
> Given the effort you put into formatting the gist (thank you) can you confirm that running this exact code you get the issue? (and not this code with some more code you haven't published).

Yeah, I have created this isolated example first to make sure I wouldn't waste your time on something I couldn't reproduce on a clean source tree.

Here's a recording of it.

https://user-images.githubusercontent.com/2079069/221660313-e27fb354-dbc6-433e-a56e-1b56d7f87704.mp4

I've added a 3-sec pause frame to the recording when it happens, but there's a tiny delay between the window and terminal due to capture card/ssh. It should be pretty obvious to see however.
